[Bryan Lim Jing Xiang] iP

bryanljx opened the pull request:
## Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. Download from [here](https://github.com/bryanljx/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you :smile:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke("data/saveFile.txt").run();
    }
}
```

## Future improvements
Work in progress:
- [ ] Improve error handling (Handle more exceptions)
- [ ] Improve test coverage
- [ ] Parse more date formats
- [ ] Improve look and feel!

Commit:
Add Gradle support

bryanljx committed:
Set up skeletal version of Duke

bryanljx committed:
Revert "Set up skeletal version of Duke"

This reverts commit cbdb45399bde50056d53a8f9d0bf6accbf5119cf.

bryanljx committed:
Set up skeletal version of Duke

bryanljx committed:
Revert "Set up skeletal version of Duke"

This reverts commit 295c10ab4886ea522fdcfd58995285b93d2c5303.

bryanljx committed:
Set up skeletal version of Duke

bryanljx committed:
Refactor text formatting to separate class

bryanljx committed:
Add list that stores/display text entered by user

bryanljx committed:
Add Task that can be marked/unmarked

bryanljx committed:
Refactor formatting of multiple lines to new function

bryanljx committed:
Tidy variables and return types in Task

bryanljx committed:
Revert "Tidy variables and return types in Task"

This reverts commit 1af53d84f38f5501ec3c3a1a865ab3234e84e8ee.

bryanljx committed:
Add Task subclasses for todo, event, deadline

bryanljx committed:
Update TaskList to match changes to Task

bryanljx committed:
Update duke to handle todo, event, deadline

bryanljx committed:
Fix regex to match / instead of "

bryanljx committed:
Update comments to match regex change

bryanljx committed:
Update test script and files to automate testing

bryanljx committed:
Add error handling

bryanljx committed:
Add delete functionality to TaskList

bryanljx committed:
Add encoding to Tasks classes and group them into task package

bryanljx committed:
Refactor TextFormatter to Ui class placed in util package

bryanljx committed:
Refactor and move TaskList to tasklist package

bryanljx committed:
Refactor DukeException and add unknown task encoding exception

bryanljx committed:
Add storage class

bryanljx committed:
Add Parser class to util pkg

bryanljx committed:
Add Command abstract class

bryanljx committed:
Move execution logic to relevant Command subclasses

bryanljx committed:
Update Duke to match previous changes

bryanljx committed:
Add data/saveFile.txt to gitIgnore

bryanljx committed:
Edit Duke to only parse /by or /at for Deadline and Event respectively

bryanljx committed:
Add invalid datetime format exceptions

bryanljx committed:
Add Date classes

bryanljx committed:
Update tasklist and tasks to use date classes

bryanljx committed:
Update Duke to match date changes

bryanljx committed:
Merge branch 'branch-Level-7'

bryanljx committed:
Merge branch 'branch-Level-8'

bryanljx committed:
Move date classes to date pkg

bryanljx committed:
Add encoding for Deadline/Event

bryanljx committed:
Fix regression in exception when resolving merge conflict

bryanljx committed:
Update Parser to proper parse Deadline/Event

bryanljx committed:
Fix Storage not creating new save file if dir exists

bryanljx committed:
Merge branch 'add-gradle-support'

bryanljx committed:
Add stubs for Task class

bryanljx committed:
Add unit tests for TaskList

bryanljx committed:
Add unit tests for Task subclasses

bryanljx committed:
Update build.gradle on main class classname

bryanljx committed:
Remove unnecessary println statement

bryanljx committed:
Remove symbol from error messages

bryanljx committed:
Add find functionality to TaskList

bryanljx committed:
Add find command

bryanljx committed:
Update Parse to parse find command

bryanljx committed:
Set up checkstyle with Gradle

bryanljx committed:
Fix coding style violations

bryanljx committed:
Add javadocs

Note that this branch is done after the A-CodingStandards when auto format on save in intelliJ was set up so there might be some files that were formatted as well.

bryanljx committed:
Merge branch 'branch-A-JavaDoc'

bryanljx committed:
Add javadocs for FindSearchTextCommand

bryanljx committed:
Merge branch 'branch-A-CodingStandard'

bryanljx committed:
Add javadocs for Find functions

bryanljx committed:
Add import for @DisplayName annotation

zzthian commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public abstract class Task {
+    private final String taskItem;
+    private boolean isMarked;
```
Good naming of boolean variables, makes it easier to follow the code 👍

zzthian commented on src/main/java/date/DeadlineDateTime.java:
```diff
@@ -0,0 +1,77 @@
+     * @return DeadlineDateTime object that stores the Date and Time
+     * @throws DukeException If the storedDateTime cannot be parsed
+     */
+    public static DeadlineDateTime parseDateFromStorage(String storedDateTime) throws DukeException {
```
Great naming of methods, clear usage of verbs 😄

zzthian reviewed (commented):
Overall very well written to comply with the Java Coding standard. Great use of access modifiers and naming was very clear. Great job! 😀

bryanljx committed:
Add javaFX to application

bryanljx committed:
Modify Command classes to store/return output message

bryanljx committed:
Add gui elements

bryanljx committed:
Update Duke and Ui according to gui changes

czhongwei commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,55 @@
+    public static String formatLinesIntoParagraph(String... lines) {
+        String res = "";
+        for (String line : lines) {
+            res += formatLine(line);
```
Since you're concatenating Strings here, maybe you could consider using the StringBuilder object to optimise your code?

czhongwei reviewed (commented):
Good work! I couldn't really find any coding standard violations

MinHeinA reviewed (commented):
Could add more unit test for more of the classes.

The packages are done neatly with the related classes placed in the same folder (package). Follows good OO practices, good naming of variables and well code is well documented.

bryanljx committed:
Merge tag 'Level-10'

bryanljx committed:
Add assertions after validating args where appropriate

Added validations after validating arguments/format
for Todo arguments, EventDateTime, and DeadlineDateTime.

bryanljx committed:
Refactor string concatenation to use StringBuilder instead

There were a few cases where String concatenation was used within
a loop instead of StringBuilder, which is not ideal due to
performance reasons. This was fixed in this commit.

bryanljx committed:
Add equals method for Date Time classes

bryanljx committed:
Add equals method for Task classes

bryanljx committed:
Implement check for duplicated tasks in TaskList

bryanljx committed:
Update AddTaskCommand to account for duplicated tasks

bryanljx committed:
Rectify error in date format for INVALID_EVENT_DATETIME_FORMAT

Date format for event datetime is `yyyy-mm-dd hh:mm:ss hh:mm:ss`,
not `yyyy-mm hh:mm:ss hh:mm:ss  hh:mm:ss`. This has been rectified
in this commit.

bryanljx committed:
Enable assertions in gradle file

bryanljx committed:
Merge pull request #1 from bryanljx/branch-A-Assertions

Add Assertions

bryanljx committed:
Merge pull request #2 from bryanljx/branch-A-CodeQuality

Improve code quality

bryanljx committed:
Merge branch 'branch-C-DetectDuplicates'

bryanljx committed:
Add screenshot of app

bryanljx committed:
Update User Guide

bryanljx committed:
Update tests

bryanljx committed:
Update build.gradle

bryanljx committed:
Update user guide

bryanljx committed:
Update gradle.build
